runtime/near-vm-logic/src/tests/gas_counter.rs

```diff
@@ -103,54 +103,103 @@ fn test_hit_prepaid_gas_limit() {
 }
 
 #[cfg(feature = "protocol_feature_function_call_weight")]
-fn function_call_weight_check(function_calls: impl IntoIterator<Item = (u64, u64)>) {
+fn function_call_weight_check(function_calls: &[(Gas, u64)]) {
+    use crate::receipt_manager::ReceiptMetadata;
+    use near_primitives::transaction::{Action, FunctionCallAction};
     use near_primitives::types::GasWeight;
 
     let gas_limit = 10u64.pow(14);
 
     let mut logic_builder = VMLogicBuilder::default().max_gas_burnt(gas_limit);
     let mut logic = logic_builder.build_with_prepaid_gas(gas_limit);
 
+    let mut ratios = vec![];
+
+    // Schedule all function calls
     for (static_gas, gas_weight) in function_calls {
         let index = promise_batch_create(&mut logic, "rick.test").expect("should create a promise");
         promise_batch_action_function_call_weight(
             &mut logic,
             index,
             0,
-            static_gas,
-            GasWeight(gas_weight),
+            *static_gas,
+            GasWeight(*gas_weight),
         )
         .expect("batch action function call should succeed");
+        ratios.push((index, *gas_weight));
+    }
+
+    fn assert_with_gas(receipt: &ReceiptMetadata, cb: impl Fn(Gas) -> bool) {
+        if let Action::FunctionCall(FunctionCallAction { gas, .. }) = receipt.actions()[0] {
+            assert!(cb(gas));
+        } else {
+            panic!("expected function call action");
+        }
     }
+
+    let (gas_per_weight, remainder_gas) = {
+        let gas_to_distribute = logic.gas_counter().remaining_gas();
+        let weight_sum: u64 = function_calls
+            .iter()
+            .map(|(_, weight)| weight)
+            // Saturating sum of weights
+            .fold(0, |sum, val| sum.checked_add(*val).unwrap_or(u64::MAX));
+        (gas_to_distribute / weight_sum, gas_to_distribute % weight_sum)
+    };
```

Comment:
Yeah, that's a good suggestion, I was avoiding doing specific values to avoid being brittle, but I will change this. (I might keep some of this for a prop based fuzz test though)